src/source/promises.bs

```diff
@@ -38,6 +38,151 @@ namespace promises
         return promises.internal.on("finally", promise, callback, context)
     end function
 
+    ' The Promise.try() method takes a callback of any kind (returns or throws, synchronously or asynchronously) and wraps its result in a Promise.
+    ' @param {function} callback - The callback to wrap in a promise
+    ' @param {dynamic} args - The arguments to pass to the callback. Max 32 arguments.
+    function try(callback as function, args = invalid as dynamic) as dynamic
+        try
+            if type(args) = "roArray" then
+                argsCount = args.count()
+
+                if argsCount < 16 then ' 0-15 args
```

Comment:
I think we should move this into a promises.internal helper function, named something like `callWithArgs()`. Then this function's logic can be much cleaner and clear what's happening.